x/observer/keeper/ballot.go

```diff
@@ -58,10 +71,25 @@ func (k Keeper) GetAllBallots(ctx sdk.Context) (voters []*types.Ballot) {
 
 // AddBallotToList adds a ballot to the list of ballots for a given height.
 func (k Keeper) AddBallotToList(ctx sdk.Context, ballot types.Ballot) {
-	list, found := k.GetBallotList(ctx, ballot.BallotCreationHeight)
+	list, found := k.GetBallotListForHeight(ctx, ballot.BallotCreationHeight)
 	if !found {
 		list = types.BallotListForHeight{Height: ballot.BallotCreationHeight, BallotsIndexList: []string{}}
 	}
 	list.BallotsIndexList = append(list.BallotsIndexList, ballot.BallotIdentifier)
 	k.SetBallotList(ctx, &list)
 }
+
+// ClearMaturedBallotsAndBallotList deletes all matured ballots and the list of ballots for a given height.
+// It also emits an event for each ballot deleted.
+func (k Keeper) ClearMaturedBallotsAndBallotList(ctx sdk.Context, ballots []types.Ballot, maturityBlocks int64) {
+	for _, ballot := range ballots {
+		k.DeleteBallot(ctx, ballot.BallotIdentifier)
+		EmitEventBallotDeleted(ctx, ballot)
```

Comment:
I don't think we need an event here

Reply:
Once indexed, the event can be used to query details about deleted ballots without having to query an archive node with an older height. 
https://github.com/zeta-chain/node/issues/942#issuecomment-2315975640

Reply:
I do agree though we should remove the event if its not useful .

Reply:
Events would go into the blockstore , and I think nodes have the option to not store them through configuration, events usually are not part of the application.db .
But yeah, we should avoid adding anything to the node that we don't need 
The primary use case that I can think of , where we use the data from these events is for debugging to see if there is a particular observer missing votes consistently fro more than maturityBlocks